### PR TITLE
fix selectStatement contained tableReferences for other db

### PR DESCRIPTION
**Pull request by jingshanglu**
ref #4885 .

Changes proposed in this pull request:
- fix selectStatement contained tableReferences for other db

### CODE REVIEW

**Comment by jingshanglu** [comment]
Test of binder module use tables of selectStatement, cannot remove  tables.  @tristaZero

File: shardingsphere-sql-parser/shardingsphere-sql-parser-statement/src/main/java/org/apache/shardingsphere/sql/parser/sql/statement/dml/SelectStatement.java [hunk 1]
```diff
@@ -41,10 +40,10 @@
 @Setter
 public final class SelectStatement extends DMLStatement {
     
-    private final Collection<TableSegment> tables = new LinkedList<>();
-    
     private ProjectionsSegment projections;
     
+    private final Collection<TableSegment> tables = new LinkedList<>();
+
```
Comment (tristaZero) [inline shardingsphere-sql-parser/shardingsphere-sql-parser-statement/src/main/java/org/apache/shardingsphere/sql/parser/sql/statement/dml/SelectStatement.java]: Sorry, `tables` was just moved below?

File: shardingsphere-sql-parser/shardingsphere-sql-parser-statement/src/main/java/org/apache/shardingsphere/sql/parser/sql/statement/dml/SelectStatement.java [hunk 2]
```diff
@@ -118,19 +117,4 @@
         }
         return result;
     }
-    
-    /**
-     * Get subQuery table segments.
-     *
-     * @return subQuery table segments
-     */
-    public Collection<SubqueryTableSegment> getSubQueryTableSegments() {
```
Comment (tristaZero) [inline shardingsphere-sql-parser/shardingsphere-sql-parser-statement/src/main/java/org/apache/shardingsphere/sql/parser/sql/statement/dml/SelectStatement.java]: Why did you remove this function?

**Comment by tristaZero** [comment]
@jingshanglu You means `tables` will be removed in the future, not now?

File: shardingsphere-sql-parser/shardingsphere-sql-parser-statement/src/main/java/org/apache/shardingsphere/sql/parser/sql/statement/dml/SelectStatement.java (hunk 2 above)
Comment (jingshanglu) [reply shardingsphere-sql-parser/shardingsphere-sql-parser-statement/src/main/java/org/apache/shardingsphere/sql/parser/sql/statement/dml/SelectStatement.java]: This func not be used.

File: shardingsphere-sql-parser/shardingsphere-sql-parser-statement/src/main/java/org/apache/shardingsphere/sql/parser/sql/statement/dml/SelectStatement.java (hunk 1 above)
Comment (jingshanglu) [reply shardingsphere-sql-parser/shardingsphere-sql-parser-statement/src/main/java/org/apache/shardingsphere/sql/parser/sql/statement/dml/SelectStatement.java]: It has recovered.

**Comment by jingshanglu** [comment]
> @jingshanglu You mean `tables` will be removed in the future, not now?

The premise is that the binder module needs to be modified. @tristaZero

**Comment by jingshanglu** [comment]
Modify the binder module test in the next subtask, how do you think? @tristaZero

**Comment by tristaZero** [comment]
Good job, and it is time to review the SQL cases below whether it could cover all the databases except for `MySQL`(They are updated  in your last PR).
> 1. select_inner_join_related_with_name
 2. select_sharding_route_with_broadcast_table
